website/docs/reference/resource-properties/arguments.md

```diff
@@ -0,0 +1,118 @@
+---
+title: arguments
+sidebar_label: "arguments"
+id: arguments
+---
+
+import MacroArgsNote from '/snippets/_validate-macro-args.md';
+
+<File name='macros/<filename>.yml'>
+
+```yml
+
+version: 2
+
+macros:
+  - name: <macro name>
+    arguments:
+      - name: <arg name>
+        type: <string>
+        description: <markdown_string>
+
+```
+
+</File>
+
+## Definition
+
+The `arguments` property is used to define the parameters that a macro can accept. Each argument can have a `name`, `type`, and `description`. This helps in documenting the macro and understanding what inputs it requires.
+
+You can define `arguments` in a [`properties.yml`](/reference/resource-properties/arguments#macro-properties) file. You can name this file anything and it can placed in the [`macro-paths`](/reference/project-configs/macro-paths) directory (defaults to `["macros"]`). 
```

Comment:
```suggestion
You can define `arguments` in a [`properties.yml`](/reference/macro-properties) file. You can name this file anything and it can placed in the [`macro-paths`](/reference/project-configs/macro-paths) directory (defaults to `["macros"]`). 
```